Add plugin: Template by Note Name

jacoblearned opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/jacoblearned/obsidian-template-by-note-name

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[# Obsidian Template by Note Name](https://github.com/jacoblearned/obsidian-template-by-note-name/blob/4d24f4d17eba9371f8a5b95b13abee559bd5a11a/README.md#L1)
Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.

[await this.app.vault.process(file, () => {](https://github.com/jacoblearned/obsidian-template-by-note-name/blob/4d24f4d17eba9371f8a5b95b13abee559bd5a11a/src/main.ts#L51-L53)
Use the `modify` function instead of process, using `process` only makes sense if you actually use the content the file had before.

[containerEl.createEl("h1", { text: "Template by Note Name" });](https://github.com/jacoblearned/obsidian-template-by-note-name/blob/4d24f4d17eba9371f8a5b95b13abee559bd5a11a/src/main.ts#L235)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

jacoblearned commented:
@joethei Thanks for the review! I've addressed the comments in the repo and cut a new 1.0.0 release with the requested changes

Changes: https://github.com/jacoblearned/obsidian-template-by-note-name/pull/50
Release: https://github.com/jacoblearned/obsidian-template-by-note-name/releases/tag/1.0.0

Please let me know if you need anything else.
